plugins/plant.cpp

```diff
@@ -0,0 +1,627 @@
+        return x >= x_min && x <= x_max &&
+            y >= y_min && y <= y_max &&
+            z >= z_min && z <= z_max;
+    }
+    inline bool testPos(df::coord pos) const { return testPos(pos.x, pos.y, pos.z); }
+};
+
+struct plant_options
+{
+    bool create = false; // Create a plant
+    bool grow = false; // Grow saplings into trees
+    bool del = false; // Remove plants
+    bool shrubs = false; // Remove shrubs
+    bool saplings = false; // Remove saplings
+    bool trees = false; // Remove grown trees
+    bool dead = false; // Remove only dead plants
+    bool filter_ex = false; // Filter vector excludes if true, else requires its plants
+    bool zlevel = false; // Operate on entire z-levels
+    bool dry_run = false; // Don't actually grow or remove anything
+
+    int32_t plant_idx = -1; // Plant raw index of plant to create; -2 means print all non-grass ids
+    int32_t age = -1; // Set plant to this age for grow/create; -1 for default
+
+    static struct_identity _identity;
+};
+static const struct_field_info plant_options_fields[] =
+{
+    { struct_field_info::PRIMITIVE, "create",    offsetof(plant_options, create),    &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "grow",      offsetof(plant_options, grow),      &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "del",       offsetof(plant_options, del),       &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "shrubs",    offsetof(plant_options, shrubs),    &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "saplings",  offsetof(plant_options, saplings),  &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "trees",     offsetof(plant_options, trees),     &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "dead",      offsetof(plant_options, dead),      &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "filter_ex", offsetof(plant_options, filter_ex), &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "zlevel",    offsetof(plant_options, zlevel),    &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "dry_run",   offsetof(plant_options, dry_run),   &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "plant_idx", offsetof(plant_options, plant_idx), &df::identity_traits<int32_t>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "age",       offsetof(plant_options, age),       &df::identity_traits<int32_t>::identity, 0, 0 },
+    { struct_field_info::END }
+};
+struct_identity plant_options::_identity(sizeof(plant_options), &df::allocator_fn<plant_options>, NULL, "plant_options", NULL, plant_options_fields);
+
+const int32_t sapling_to_tree_threshold = 120 * 28 * 12 * 3 - 1; // 3 years minus 1; let the game handle the actual growing-up
```

Comment:
it's silly that we don't have constants for these already. something we should think about when we address https://github.com/DFHack/dfhack/pull/3907